vowpalwabbit/global_data.h

```diff
@@ -359,6 +359,8 @@ struct vw
   AllReduceType all_reduce_type;
   AllReduce* all_reduce;
 
+  bool chain_hash = false;
```

Comment:
[Minor] Since this seems to only be consumed by the parser, shoudl this live on the `parser* p` object above (like p->hasher), see parser_args.cc:670 in your branch.